generator/internal/dart/generate.go

```diff
@@ -26,15 +27,25 @@ import (
 //go:embed templates
 var dartTemplates embed.FS
 
-func Generate(model *api.API, outdir string, cfg *config.Config) error {
+func Generate(model *api.API, outdir string, config *config.Config) error {
 	annotate := newAnnotateModel(model)
-	_, err := annotate.annotateModel(cfg.Codec)
+	_, err := annotate.annotateModel(config.Codec)
 	if err != nil {
 		return err
 	}
+
 	provider := templatesProvider()
-	// TODO(#1564): Walk the generated files; 'dart format' Dart ones.
-	return language.GenerateFromRoot(outdir, model, provider, generatedFiles(model))
+	err = language.GenerateFromRoot(outdir, model, provider, generatedFiles(model))
+	if err == nil {
+		// Check if we're configured to skip formatting.
+		skipFormat := config.Codec["skip-format"]
```

Comment:
this is clever, it neatly bypasses all the problems with the different CI builds needing to have the full toolchain.

Reply:
Thanks! I'd considered allowing the format to fail w/o returning an error, but the tests would likely still have problems (local developers would be able to format; some CIs would not).